code/addons/test/src/vitest-plugin/index.ts

```diff
@@ -82,56 +96,124 @@ export const storybookTest = (options?: UserOptions): Plugin => {
-
-      for (const staticDir of staticDirs) {
-        try {
-          statics.push(mapStaticDir(staticDir, configDir));
-        } catch (e) {
-          console.warn(e);
-        }
-      }
 
       // If we end up needing to know if we are running in browser mode later
       // const isRunningInBrowserMode = config.plugins.find((plugin: Plugin) =>
       //   plugin.name?.startsWith('vitest:browser')
       // )
 
-      config.test ??= {};
+      const baseConfig: Omit<ViteUserConfig, 'plugins'> = {
+        test: {
+          setupFiles: ['@storybook/experimental-addon-test/internal/setup-file'],
+
+          ...(finalOptions.storybookScript
+            ? {
+                globalSetup: ['@storybook/experimental-addon-test/internal/global-setup'],
+              }
+            : {}),
+
+          env: {
+            ...storybookEnv,
+            // To be accessed by the setup file
+            __STORYBOOK_URL__: storybookUrl,
+            __VITEST_INCLUDE_TAGS__: finalOptions.tags.include.join(','),
+            __VITEST_EXCLUDE_TAGS__: finalOptions.tags.exclude.join(','),
+            __VITEST_SKIP_TAGS__: finalOptions.tags.skip.join(','),
+          },
+
+          include: storiesFiles
+            .filter((path) => !path.endsWith('.mdx'))
+            .map((path) => convertPathToPattern(path)),
+
+          server: {
+            deps: {
+              inline: ['@storybook/experimental-addon-test'],
+            },
+          },
+
+          ...(input.test.browser
+            ? {
+                browser: {
+                  name: input.test.browser.name,
+                  screenshotFailures: input.test.browser?.screenshotFailures ?? false,
+                },
+              }
+            : {}),
+        },
+
+        envPrefix: ['STORYBOOK_', 'VITE_'],
+
+        // copying straight from https://github.com/vitejs/vite/blob/main/packages/vite/src/node/constants.ts#L60
+        // to avoid having to maintain Vite as a dependency just for this
+        resolve: {
+          conditions: [
+            'storybook',
+            'stories',
+            'test',
+            'module',
+            'browser',
+            'development|production',
+          ],
+        },
+
+        optimizeDeps: {
+          include: [
+            '@storybook/experimental-addon-test/**',
+            ...(frameworkName?.includes('react') || frameworkName?.includes('nextjs')
+              ? ['react-dom/test-utils']
+              : []),
+          ],
+        },
+
+        define: {
+          ...(frameworkName?.includes('vue3')
+            ? { __VUE_PROD_HYDRATION_MISMATCH_DETAILS__: 'false' }
+            : {}),
+        },
+      };
+
+      // Merge config from storybook with the plugin config
+      const config: Omit<ViteUserConfig, 'plugins'> = mergeConfig(
+        baseConfig,
+        viteConfigFromStorybook
+      );
 
+      // alert the user of problems
       if (config.test.include?.length > 0) {
```

Comment:
`config` now doesn't include anything from the user's input, so we need to change the check here I think.

```suggestion
      // alert the user of problems
      if (input.test.include?.length > 0) {
```